modules/bluefish/consumer/bluefish_consumer.cpp

```diff
@@ -237,15 +455,85 @@ struct bluefish_consumer : boost::noncopyable
 		});
 	}
 
-	void display_frame(core::const_frame frame)
+	static void dma_present_thread_actual(void* arg)
 	{
-		// Sync
+		bluefish_consumer* blue = (bluefish_consumer*)arg;
 
-		sync_timer_.restart();
-		unsigned long n_field = 0;
-		blue_->wait_output_video_synch(UPD_FMT_FRAME, n_field);
-		graph_->set_value("sync-time", sync_timer_.elapsed()*format_desc_.fps*0.5);
-		
+		bvc_wrapper wait_b;
+		wait_b.attach(blue->device_index_);
+		EBlueVideoChannel out_vid_channel = get_bluesdk_videochannel_from_streamid(blue->device_output_channel_);
+		wait_b.set_card_property32(DEFAULT_VIDEO_OUTPUT_CHANNEL, out_vid_channel);
+		int frames_to_buffer = 3;
```

Comment:
I tried 1 here and I still got no dropped frames when running the 3o firmware with SDI A, B and C. Did you see a need for more than 1?